src/_app/render.tsx

```diff
@@ -11,21 +10,31 @@ export default function renderApp({
   options: ExtendedOptions;
   pageObject: PageObject;
   pageData: PageData;
-}) {
-  const { useApp } = options;
-  const customAppFile = getCustomAppFile({ options });
-  let AppComponent;
+}): JSX.Element {
+  const { env } = options;
+  const { appFile } = pageObject;
+  const AppComponent = appFile[env].default;
 
-  if (useApp && customAppFile?.client?.default) {
-    AppComponent = customAppFile.client.default;
-  } else {
-    AppComponent = DefaultApp;
-  }
-
-  return (
+  // @NOTE This generate "multiple React instances" error
+  return env === 'server' ? (
+    executeAsIfOnServerSync(() => (
```

Comment:
I think we don't need to wrap it with `executeAsIfOnServerSync` here and I assume that is causing the problems. We are not rendering here, but just wrapping JSX.Element, which is of correct environment already.

Reply:
If I get the whole picture, we need `executeAsIfOnServerSync` here to ensure that components are executed (not loaded) in `server` environment (eg. getting the right result from `typeof window === 'undefined'`).

I found out that `jest.requireActual` works as expected when the module to be requires gets required before the `jest.requireActual` call 🤔 . Maybe another Jest bug?

Reply:
BTW once we add tests for this change we'll be able to dispel any doubt.

Reply:
By execution do you mean being rendered? So we have 3 lifecycle "phases" of each "component":
- import time ✅ 
- data fetching time ✅ 
- render time (this happens after -- when we are rendering the component -- `renderToStaticMarkup`  or `rtl.render`)

I might be wrong, but at this point we are just wrapping the component so I think its redundant. Anyhow, don't we have tests for global `_app` that would confirm my suspision?

Reply:
I guess you're right @Meemaw :)

Reply:
We have tests global-object tests, but we're not testing again SSR result because the current API doesn't expose SSRed html.